firefox_ui_harness/runners/base.py

```diff
@@ -39,9 +48,42 @@
     'focusmanager.testmode': True,
     'geo.provider.testing': True,
     'javascript.options.showInConsole': True,
+    'marionette.logging': False,
     'security.notification_enable_delay': 0,
     'signon.rememberSignons': False,
     'startup.homepage_welcome_url': 'about:blank',
     'toolkit.startup.max_resumed_crashes': -1,
     'toolkit.telemetry.enabled': False,
 }
+
+
+class FirefoxUITestRunner(BaseMarionetteTestRunner):
+    def __init__(self, **kwargs):
+        BaseMarionetteTestRunner.__init__(self, **kwargs)
+
+        if not self.server_root:
+            self.server_root = firefox_ui_tests.resources
+
+        # Set default preferences for Firefox UI tests
+        # Bug 1141129 - Workaround necessary given that self.prefs defaults to None
+        prefs = self.prefs or {}
+        prefs.update(DEFAULT_PREFS)
+        self.prefs = prefs
```

Comment:
Then the `--e10s` option for marionette is misbehaving given that it gives you totally inconsistent states. If that option is specified Firefox should run with multiple processes, otherwise not. I think what is really necessary here is a fix in marionette client: https://dxr.mozilla.org/mozilla-central/source/testing/marionette/client/marionette/runner/base.py#473

If setting the pref to False in the client is not wanted, the option should be renamed to `--force-e10s`. As of now it is totally misleading about the underlying behavior. Exactly that's why my confusion.

I will check how to make an exception for this preference but I stay with the point that our preferences have to overrule preferences as set by Marionette.